test/foundry-tests/PoolManager.t.sol

```diff
@@ -984,6 +986,153 @@ contract PoolManagerTest is Test, Deployers, TokenFixture, GasSnapshot, IERC1155
         snapEnd();
     }
 
+    function testPoolManagerNoOp(uint160 sqrtPriceX96) public {
+        // Assumptions tested in Pool.t.sol
+        vm.assume(sqrtPriceX96 >= TickMath.MIN_SQRT_RATIO);
+        vm.assume(sqrtPriceX96 < TickMath.MAX_SQRT_RATIO);
+
+        address payable hookAddr = payable(
+            address(
+                uint160(
+                    Hooks.BEFORE_INITIALIZE_FLAG | Hooks.BEFORE_MODIFY_POSITION_FLAG | Hooks.BEFORE_SWAP_FLAG
+                        | Hooks.BEFORE_DONATE_FLAG
+                )
+            )
+        );
+
+        vm.etch(hookAddr, vm.getDeployedCode("NoOpTestHooks.sol:NoOpTestHooks"));
+
+        PoolKey memory key =
+            PoolKey({currency0: currency0, currency1: currency1, fee: 100, hooks: IHooks(hookAddr), tickSpacing: 10});
+
+        // Test initialize with balance delta
+        MockERC20(Currency.unwrap(currency0)).approve(hookAddr, 100 ether);
+        MockERC20(Currency.unwrap(currency1)).approve(hookAddr, 100 ether);
+
+        NoOpTestHooks(hookAddr).initialize(manager, key, SQRT_RATIO_1_1);
+        assertEq(manager.balanceOf(hookAddr, CurrencyLibrary.toId(currency0)), 1 ether);
+
+        // Test add liquidity
+        snapStart("modify position with noop");
+        modifyPositionRouter.modifyPosition(key, IPoolManager.ModifyPositionParams(-120, 120, 10 ether));
+        snapEnd();
+
+        assertEq(manager.balanceOf(hookAddr, CurrencyLibrary.toId(currency0)), 11 ether);
+        assertEq(manager.balanceOf(hookAddr, CurrencyLibrary.toId(currency1)), 10 ether);
+
+        // Swap
+        IPoolManager.SwapParams memory swapParams =
+            IPoolManager.SwapParams({zeroForOne: true, amountSpecified: 100, sqrtPriceLimitX96: SQRT_RATIO_1_2});
+
+        PoolSwapTest.TestSettings memory testSettings =
+            PoolSwapTest.TestSettings({withdrawTokens: true, settleUsingTransfer: true});
+
+        snapStart("swap with noop");
+        BalanceDelta delta = swapRouter.swap(key, swapParams, testSettings);
+        snapEnd();
+
+        assertEq(delta.amount0(), 1 ether);
+        assertEq(delta.amount1(), -1 ether);
+
+        // Remove liquidity
+        modifyPositionRouter.modifyPosition(key, IPoolManager.ModifyPositionParams(-120, 120, -1 ether));
+
+        assertEq(manager.balanceOf(hookAddr, CurrencyLibrary.toId(currency0)), 11 ether);
+        assertEq(manager.balanceOf(hookAddr, CurrencyLibrary.toId(currency1)), 8 ether);
+
+        // Donate
+        donateRouter.donate(key, 0, 0);
+
+        // should have donated 1 eth based on hook donate logic
+        assertEq(manager.balanceOf(hookAddr, CurrencyLibrary.toId(currency0)), 12 ether);
+        assertEq(manager.balanceOf(hookAddr, CurrencyLibrary.toId(currency1)), 9 ether);
+    }
+
+    function testPoolManagerGracefulReturnsOnHook(uint160 sqrtPriceX96) public {
```

Comment:
I'd say this description does not explain what the test is testing, but just saying that it's testing a test contract. IMO tests should generally be very targeted (like the majority of the tests in this file) and only in certain special cases a test can be very comprehensive and go through many many things. I'm not completely following this whole GracefulReturns style testing.... it seems a bit confusing and unreadable to me